File: js/p0001/p0001.js
export default function findSum(nums, target) {
  const len = nums.length;
  const s = new Map();

  for (let i = 0; i < len; i += 1) {
    const a = nums[i];
    const index = s.get(target - a);
    if (index !== undefined) {
      return [index, i];
    }
    s.set(a, i);
  }
  return undefined;
}

File: py/p0001/p0001.py
# coding=utf-8


class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mark = {}
        for index, num in enumerate(nums):
            solve = target - num
            if solve in mark:
                return [mark[solve], index]
            else:
                mark[num] = index
        return []
